react-ui/src/components/LogInForm.js

```diff
@@ -34,73 +35,90 @@ class logInForm extends Component {
     });
   };
 
-
   login = event => {
     let email = this.state.loginParams.email;
     let password = this.state.loginParams.password;
-    var params = {email: email, password: password}
-    axios.get(`/public/login/?${queryString.stringify(params)}`)
-    .then(res => {
-      if(res.data!==""){
-        this.setState({
-          email : res.data.email,
-          userType: res.data.userType,
-          firstName: res.data.firstName,
-          personId: res.data.workerId || res.data.studentId,
-          accessToken: res.data.accessToken,
-          islogged: true
+    let validEmail = false;
+
+    if (/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))*$/.test(email)) {
+      validEmail = true; // email validation so that a user can't just enter asdasdad@fsdsf and log in to blank page
+    }
+    if (validEmail === false) {
+      alert("You have entered an invalid email address!")
+    } else {
+      var params = { email: email, password: password }
+     
+      axios.get(`/public/login/?${queryString.stringify(params)}`)
+        .then(res => {
+          console.log(res.data)
+          if (_.isNil(res.error) && res.data) {
+            this.setState({
+              email: res.data.email,
+              userType: res.data.userType,
+              firstName: res.data.firstName,
+              personId: res.data.workerId || res.data.studentId,
+              accessToken: res.data.accessToken,
+              islogged: true
+            })
+            // localStorage.setItem("token", "T");
+            event.preventDefault();
+            console.log(this.state.userType)
+          } else {
+            alert("Invalid Email or Password")
+            event.target.reset();
+            event.preventDefault();
```

Comment:
Hi Kyle,
Line 64 and line 69 are redundant due to line 72, as `event.preventDefault()` would execute for both conditions already. I would suggest removing it from line 64 and line 69.

Reply:
lines 64 and 69 are removed